lm_eval/models/optimum_lm.py

```diff
@@ -0,0 +1,60 @@
+from pathlib import Path
+from lm_eval.api.registry import register_model
+from lm_eval.models.huggingface import HFLM
+
+
+@register_model("optimum-causal")
+class OptimumLM(HFLM):
+    """
+    Optimum Intel provides a simple interface to optimize Transformer models and convert them to \
+    OpenVINO™ Intermediate Representation (IR) format to accelerate end-to-end pipelines on \
+    Intel® architectures using OpenVINO™ runtime.
+    """
+
+    def __init__(
+        self,
+        device = "cpu",
+        **kwargs,
+    ) -> None:
+        
+        if "backend" in kwargs:
+            # optimum currently only supports causal models.
+            assert kwargs["backend"] == "causal"
+        else:
+            raise Exception("Please be sure your model is a `causal` model.")
+
+        assert device == "cpu"
```

Comment:
```suggestion
        
```

Following @ngaloppo 's comment, this should actually be dropped I believe. User can pass either CPU or Cuda, but we'll default to CPU

Reply:
What are we expecting to happen with `device = cuda`? OpenVINO doesn't support CUDA, so we likely would have to throw an exception. However, OpenVINO supports (Intel) GPUs. Does lm-evaluation-harness allow for device strings beyond `cpu` and `cuda`?

Reply:
I think I got confused by these lines in the `master` OptimumLM: https://github.com/EleutherAI/lm-evaluation-harness/blob/0bf683b4e6a9df359b3156ba9ba8d62bdd47e0c0/lm_eval/models/gpt2.py#L201-L212 where it seems like passing cuda as a device string was still supported. 

In that case, it looks like we should support either the `"cpu"`, or `"gpu"` strings, defaulting to "cpu": the HF docs say this: 

> To run inference on Intel integrated or discrete GPU, use .to("gpu"). On GPU, models run in FP16 precision by default. (See [OpenVINO documentation](https://docs.openvino.ai/nightly/openvino_docs_install_guides_configurations_for_intel_gpu.html) about installing drivers for GPU inference).

Reply:
I modified the code to get cpu or GPU for optimum_lm.